src/class.c

```diff
@@ -909,12 +911,31 @@ mrb_block_given_p(mrb_state *mrb)
     +:      Request a not frozen object; However, except nil value
  */
 MRB_API mrb_int
-mrb_get_args(mrb_state *mrb, const char *format, ...)
+mrb_get_args(mrb_state *mrb, mrb_args_format format, ...)
+{
+  va_list ap;
+  va_start(ap, format);
+  mrb_int rc = mrb_get_args_v(mrb, format, NULL, ap);
+  va_end(ap);
+  return rc;
+}
+
+MRB_API mrb_int
+mrb_get_args_a(mrb_state *mrb, mrb_args_format format, void** args)
+{
+  va_list ap;
+  mrb_int rc = mrb_get_args_v(mrb, format, args, ap);
+  return rc;
+}
+
+#define GET_ARG(_type) (ptr ? ((_type)(*ptr++)) : va_arg(ap, _type))
+
+MRB_API mrb_int
+mrb_get_args_v(mrb_state *mrb, mrb_args_format format, void** ptr, va_list ap)
```

Comment:
having an `MRB_API` (public API) with a `va_list` parameter will make it difficult to provide an API and ABI compatible MRB API implementation outside of C. Is it intentional that this function is marked as part of the public API?